src/domain/index.ts

```diff
@@ -69,6 +66,12 @@ function handleGPProposalPrivate(proposalId: string ): void {
        gpProposal.submittedTime,
      );
      insertGPRewardsToHelper(gpProposal.proposalId, gpProposal.proposer as Address);
+     handleStateChange(
```

Comment:
how about call  updateProposalState unseen of handleStateChange ?
It will be more optimal code.